### ISSUE TITLE
Need an option to disable home directory creation when host user creation is used

### ISSUE THREAD

**Issue by jentfoo** [gravitational/teleport#32483]
When using [host user creation](https://goteleport.com/docs/server-access/guides/host-user-creation/) we currently offer very limited configuration options.  Some customers are using this feature in a very ephemeral way.  As such they would like to avoid the creation of a home directory when the user is added to the system.

**Comment by zmb3** [comment]
@jentfoo this is no longer applicable, correct?

**Comment by jentfoo** [comment]
@zmb3 I will defer to your preference.  This is a `feature-request`, and I believe still has some value in encouraging a more ephemeral type of usage with this feature.  That said, this is not designed to address a direct security risk.

The argument possibly for closing this is we still should consider this feature in a deeper manner.  Host user creation does expose some security risks regardless of the inclusion of this option: https://github.com/gravitational/teleport-private/issues/1028

**Comment by zmb3** [comment]
If my understanding is correct, we do have the option to disable home directory creation, `create_host_user_mode: insecure_drop`, but we advise against it for security reasons. Is that right?

**Comment by jentfoo** [comment]
@zmb3 That's not correct.  `insecure_drop` is the option which will remove the user after the session ends, and is the option which exposes a user to the [linked issue](https://github.com/gravitational/teleport-private/issues/1028) due to the potential of UID reuse.

This option would improve the normal experience (not using `insecure_drop`) by not having a bunch of unused home directories (though the users would remain on the system).

Let me know if you have any questions, happy to discuss in DM as well
